proposals/0000-wire-protocol.md

```diff
@@ -0,0 +1,435 @@
+
+Title: **DEP-0000: Hypercore Wire Protocol**
+
+Short Name: `0000-wire-protocol`
+
+Type: Standard
+
+Status: Undefined (as of 2018-02-04)
+
+Github PR: (add HTTPS link here after PR is opened)
+
+Authors: [Paul Frazee](https://github.com/pfrazee), [Bryan Newbold](https://github.com/bnewbold)
+
+
+# Summary
+[summary]: #summary
+
+This DEP describes the Hypercore wire protocol: a transport-agnostic message stream spoken between nodes in a swarm of hypercore network peers (including Dat clients). The wire protocol includes mechanisms for framing, stream encryption, and feed key authentication.
+
+
+# Motivation
+[motivation]: #motivation
+
+The protocol described here is already in use as of 2017 (by Hypercore, Dat, and Beaker Browser users), and was partially described in an earlier [whitepaper][whitepaper]. This document fills in some additional details.
+
+[whitepaper]: https://github.com/datprotocol/whitepaper
+
+
+# Stream Connections
+[stream-details]: #stream-details
+
+The Dat wire protocol depends on the use of a binary transport protocol which provides the following semantics:
```

Comment:
Should we give TCP and uTP as examples here?

Reply:
Good idea